### PR TITLE
Default selection for dialog (fixes #258)

### LINKED ISSUE #258: Default selection on Confirm dialog component
When selecting 'delete' from the work item menu on the work item list a 'confirm dialog' pops up. 

One of the options should be 'default' selected and respond to keyboard input.

Either 'Cancel' is selected and you can do Arrow-Left + Enter to move to confirm and perform the action Or 'Confirm' is selected and you do Enter to confirm and perform the action.

And 'Esc' should be same as 'Cancel'

### PR DESCRIPTION
**Pull request by sanbornsen**
fixes #258

### CODE REVIEW

**Comment by joshuawilson** [comment]
@sanbornsen please rebase

File: src/app/shared-component/dialog/dialog.component.ts [hunk 1]
```diff
@@ -39,4 +40,22 @@ export class DialogComponent implements OnInit {
     this.onClick.emit(val);
     this.modalFadeIn = false;
   }
+
+  @HostListener('window:keydown', ['$event'])
+  onActionKeyPress(event: any): void {
+    let currentDefaultIndex = this.dialog.actionButtons.findIndex(i => i.default);
```
Comment (joshuawilson) [inline src/app/shared-component/dialog/dialog.component.ts]: What is "default" here? I'm getting a Type error on it each time it is used in this method.
Error:(79, 74) TS2339:Property 'default' does not exist on type 'Object'.

Comment (sanbornsen) [inline src/app/shared-component/dialog/dialog.component.ts]: Default is a new property to the dialog action data. Please refer to this file src/app/shared-component/dialog/dialog.ts. You should not get the error.

Comment (joshuawilson) [inline src/app/shared-component/dialog/dialog.component.ts]: The reason for the error is that the class ActionItem is a private class and it's Type can't be seen by TypeScript. This makes TS think that "default" is of Type Object. To fix this, you just need to "export" the class.

**Pull request merged**